wire-compiler/src/main/java/com/squareup/wire/schema/NewSchemaLoader.kt

```diff
@@ -148,6 +152,76 @@ class NewSchemaLoader(
   override fun close() {
     return closer.close()
   }
+
+  override fun loadProfile(name: String, schema: Schema): Profile {
+    val allLocations = schema.protoFiles.map { it.location }
+    val locationsToCheck = locationsToCheck(name, allLocations)
+
+    val profileElements = mutableListOf<ProfileFileElement>()
+    for (location in locationsToCheck) {
```

Comment:
I think it could be replaced with a `mapNotNull` and a `flatMap` :)

Reply:
I am uncertain we can do that is this case, because we need to variable used in the `mapNotNull` in the `flatMap`, but I might be miss understanding